src/SpectatorView.Unity/Assets/SpectatorView/Scripts/WorldAnchors/WorldAnchorCoordinateService.cs

```diff
@@ -0,0 +1,267 @@
+        {
+        }
+        
+        protected override bool SupportsDiscovery => false;
+
+        public static Task<WorldAnchorCoordinateService> GetSharedCoordinateServiceAsync()
+        {
+            if (sharedCoordinateService == null)
+            {
+                sharedCoordinateService = InitializeSharedCoordinateServiceAsync();
+            }
+
+            return sharedCoordinateService;
+        }
+
+        private static async Task<WorldAnchorCoordinateService> InitializeSharedCoordinateServiceAsync()
+        {
+            WorldAnchorCoordinateService service = new WorldAnchorCoordinateService();
+            await service.InitializeKnownCoordinatesAsync();
+            return service;
+        }
+
+        protected override Task OnDiscoverCoordinatesAsync(CancellationToken cancellationToken, string[] idsToLocate = null)
+        {
+            throw new NotSupportedException("WorldAnchorCoordinateService does not support discovery");
+        }
+
+        protected override bool TryParse(string id, out string result)
+        {
+            result = id;
+            return true;
+        }
+
+        public async Task InitializeKnownCoordinatesAsync()
+        {
+#if UNITY_WSA
+            WorldAnchorStore store = await GetWorldAnchorStoreAsync();
+
+            foreach (string knownId in store.GetAllIds())
+            {
+                GameObject anchorGameObject = new GameObject(knownId);
+                WorldAnchor anchor = store.Load(knownId, anchorGameObject);
+
+                if (anchor == null)
+                {
+                    Debug.LogError($"Unexpected WorldAnchor {knownId} was enumerated but failed to load");
+                    continue;
+                }
+
+                OnNewCoordinate(knownId, new WorldAnchorSpatialCoordinate(knownId, anchor));
+            }
+#else
+            await Task.CompletedTask;
+#endif
+        }
+
+        public async Task<ISpatialCoordinate> CreateCoordinateAsync(string id, Vector3 worldPosition, Quaternion worldRotation, CancellationToken cancellationToken)
+        {
+#if UNITY_WSA
+            await TryDeleteCoordinateAsync(id, cancellationToken);
+
+            GameObject gameObject = new GameObject(id);
+            gameObject.transform.position = worldPosition;
+            gameObject.transform.rotation = worldRotation;
+            WorldAnchor anchor = gameObject.AddComponent<WorldAnchor>();
+
+            if (anchor.isLocated)
+            {
+                WorldAnchorStore store = await GetWorldAnchorStoreAsync();
+                store.Save(id, anchor);
+            }
+            else
+            {
+                anchor.OnTrackingChanged += Anchor_OnTrackingChanged;
+            }
+
+            WorldAnchorSpatialCoordinate coordinate = new WorldAnchorSpatialCoordinate(id, anchor);
+            OnNewCoordinate(id, coordinate);
+            return coordinate;
+#else
+            return await Task.FromResult<ISpatialCoordinate>(null);
+#endif
```

Comment:
for non UNITY_WSA instances, we should log errors stating that the current platform isn't supported.